### PR TITLE
[ENH] Added MSTL to STLForecaster

### LINKED ISSUE #240: [ENH] Add multiple seasonality support to STLForecaster
`statsmodels` is stuck with release of the `MSTL` but its already in develop and we could just copy that over or think about how else to get multiple seasonality approach implemented into `STLForecaster`.

here it is:
https://github.com/statsmodels/statsmodels/blob/6b66b2713c408483dfeb069212fa57c0ee1e078b/statsmodels/tsa/stl/mstl.py

### PR DESCRIPTION
**Pull request by aiwalter**
closes #240

### CODE REVIEW

**Review by ltsaprounis — COMMENTED** [review]
Found a couple of typos. Otherwise looks good. 

How does it work if you have multiple seasonalities? e.g. if you have hourly data and you want to model a daily and a weekly seasonality (24 and 24*7), to get the sum of the 2 seasonal components you need to use `NaiveForecaster(sp=24*7)`? If that's the case, can we add this as a tip in the docstring?

On another note, when using STL people usually forecast the signal that's trend+residuals with one model e.g. ETS(AAN) or a non-seasonal ARIMA and then add the seasonal stuff on top. I think we should make that option feasible here too, but that's another PR for another day. What do you think?

File: aeon/forecasting/trend.py [hunk 1]
```diff
@@ -464,6 +486,9 @@ def __init__(
         self.low_pass_jump = low_pass_jump
         self.inner_iter = inner_iter
         self.outer_iter = outer_iter
+        self.windows = windows
+        self.lmbda = lmbda
```
[the remark this answers is not in the document]
Comment (aiwalter) [reply aeon/forecasting/trend.py:491]: yeah seems to be a type in `statsmodels`, I can rename it for us anyways

[the remark this answers is not in the document]
Comment (aiwalter) [reply aeon/forecasting/trend.py:491]: Ok I get why thy name it `lmbda`, bcz `lambda` is a Python keyword!

**Comment by aiwalter** [comment]
thx for review, can you pls have another look. I added now a `ColumnsEnsembleForecaster` for default season forecast and an `EnsembleForecaster` for default resid forecast.

> On another note, when using STL people usually forecast the signal that's trend+residuals with one model e.g. ETS(AAN) or a non-seasonal ARIMA and then add the seasonal stuff on top. I think we should make that option feasible here too, but that's another PR for another day. What do you think?

Yes that would probably best be a separate forecaster?

File: aeon/forecasting/trend.py [hunk 2]
```diff
@@ -630,6 +699,10 @@ def get_test_params(cls, parameter_set="default"):
                 "forecaster_seasonal": NaiveForecaster(sp=3),
                 "forecaster_resid": NaiveForecaster(strategy="mean"),
             },
+            # test MSTL with list of sp
```
Comment (GuzalBulatova) [inline aeon/forecasting/trend.py:702]: Flagging these commented out lines in case they weren't left here by design

**Review by GuzalBulatova — APPROVED** [review]
One minor non-blocking comment. Amazing work, @aiwalter , looks ready to go to me!

**Comment by lmmentel** [comment]
Happy to have a closer look once the tests are passing.

**Comment by MatthewMiddlehurst** [comment]
Worth remembering that tests only happen on certain OS/Py Version currently. Its possible the failing tests arent being run on the passing actions.

**Comment by aiwalter** [comment]
oh why is that?

**Comment by MatthewMiddlehurst** [comment]
Part of the "matrix testing" implemented in sktime ages ago im pretty sure

**Comment by TonyBagnall** [comment]
I assume it was because a full test will take too long, but I wasnt involved at that time (https://github.com/aeon-toolkit/aeon/issues/154)

**Pull request closed without merging**